web/src/store/integration.js

```diff
@@ -1,19 +1,40 @@
-import { observable, action } from 'mobx';
-
-// useStrict(true);
+import { observable, action, keys } from 'mobx';
+import fetchApi from '../api/index.js';
 
 class Integration {
   @observable integrationList = [];
+  @observable groupIntegrationList = {
+    SonarQube: [],
+    SCM: [],
+    DockerRegistry: [],
+  };
+  @observable groupKeys = [];
+  @observable listLoading = true;
+  @action.bound
+  getIntegrationList() {
+    fetchApi.fetchIntegrationList({}).then(data => {
+      this.integrationList = data;
+      this.listLoading = false;
+      _.forEach(data, o => {
+        const type = _.get(o, 'spec.type');
+        if (!this.groupIntegrationList[type]) {
+          this.groupIntegrationList[type] = [o];
```

Comment:
for performance: generate a new object, and then set `this.groupIntegrationList` after loop